packages/ui/src/components/NumberInputV2/index.tsx

```diff
@@ -0,0 +1,424 @@
+    background: ${({ theme }) => theme.colors.neutral.backgroundWeak};
+    cursor: not-allowed;
+    box-shadow: none;
+  }
+
+  &[data-disabled='true'] {
+    border-color: ${({ theme }) => theme.colors.neutral.borderDisabled};
+    background: ${({ theme }) => theme.colors.neutral.backgroundDisabled};
+    cursor: not-allowed;
+  }
+`
+
+type NumberInputProps = {
+  size?: Sizes
+  /**
+   * Text displayed into component at the right of number value.
+   */
+  unit?: string
+  tooltip?: string
+  className?: string
+  'data-testid'?: string
+  label?: string
+  /**
+   * Label description displayed right next to the label. It allows you to customize the label content.
+   */
+  labelDescription?: ReactNode
+  error?: string
+  success?: string
+  helper?: ReactNode
+  value?: string | number
+} & Pick<
+  InputHTMLAttributes<HTMLInputElement>,
+  | 'onChange'
+  | 'onFocus'
+  | 'onBlur'
+  | 'name'
+  | 'id'
+  | 'placeholder'
+  | 'aria-label'
+  | 'min'
+  | 'max'
+  | 'disabled'
+  | 'step'
+  | 'readOnly'
+  | 'required'
+  | 'autoFocus'
+>
+
+/**
+ * NumberInputV2 component is used to increment / decrement a number value by clicking on + / - buttons or
+ * by typing into input. If the value is out of the min / max range, the input will automatically be the min / max value on blur.
+ */
+export const NumberInputV2 = forwardRef(
+  (
+    {
+      disabled = false,
+      max = Infinity,
+      min = 0,
+      name,
+      onChange,
+      onFocus,
+      onBlur,
+      size = 'medium',
+      step,
+      unit,
+      value,
+      tooltip,
+      className,
+      label,
+      labelDescription,
+      id,
+      placeholder = '',
+      error,
+      success,
+      helper,
+      'aria-label': ariaLabel,
+      'data-testid': dataTestId,
+      required,
+      autoFocus,
+      readOnly,
+    }: NumberInputProps,
+    ref: ForwardedRef<HTMLInputElement>,
+  ) => {
+    const localRef = createRef<HTMLInputElement>()
+    useImperativeHandle(ref, () => localRef.current as HTMLInputElement)
+
+    const uniqueId = useId()
+    const localId = id ?? uniqueId
+
+    const createChangeEvent = (
+      newValue: string | number,
+    ): ChangeEvent<HTMLInputElement> =>
+      ({
+        target: {
+          value: newValue,
+        },
+      }) as ChangeEvent<HTMLInputElement>
+
+    const onClickSideButton = useCallback(
+      (direction: 'up' | 'down') => () => {
+        if (direction === 'up') {
+          localRef.current?.stepUp()
+          onChange?.(createChangeEvent(localRef.current?.value ?? min))
+        }
+
+        if (direction === 'down') {
+          localRef.current?.stepDown()
+          onChange?.(createChangeEvent(localRef.current?.value ?? min))
+        }
+      },
+      [localRef, min, onChange],
+    )
+
+    const helperSentiment = useMemo(() => {
+      if (error) {
+        return 'danger'
+      }
+
+      if (success) {
+        return 'success'
+      }
+
+      return 'neutral'
+    }, [error, success])
```

Comment:
```suggestion
    const helperSentiment: Sentiment = useMemo(() => {
      if (error) {
        return 'danger'
      }

      if (success) {
        return 'success'
      }

      return 'neutral'
    }, [error, success])
```

?

Reply:
May be we can add some types

Reply:
@DorianMaliszewski Im not sure what you want to do, the typing is quite good already and there is no sentiments props on that component. Only success, error and helper as direct prop.

![Screenshot 2024-01-08 at 14 10 23](https://github.com/scaleway/ultraviolet/assets/15812968/4ccde872-8f67-407e-b12d-db6ab593762e)